BritneyS/DnDCharacterGenerator
# Create UI

## What does this PR do :question:
This PR contains changes related to creating and refining the UI. The functionality of the UI includes:
- Populating labels with the `Class` data returned from the request to the GraphQL server
- Returning a random `Class` in that label when the 'Generate Character' button is tapped
- Retrying the data fetch upon tapping the 'Generate Character' button if the data fetch failed the first time 


## How to test it :microscope:
⚙️**Precondition: The [dnd-graphql](https://github.com/BritneyS/dnd-graphql) server should be NOT be running on http://localhost:8000**
1. Checkout this branch
2. Run `pod install`
3. Run the app
4. Tap the 'Generate Character' button **(Important: The dnd-graphql server should not be running)**
5. Note that the labels on the screen remain blank, and the following error is printed to the console (expected!):
```
Error in fetching response: Optional(Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSUnderlyingError=0x600000249bf0 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, NSErrorFailingURLStringKey=http://localhost:8000/graphql, NSErrorFailingURLKey=http://localhost:8000/graphql, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=61, NSLocalizedDescription=Could not connect to the server.})
```
6. Run the dnd-graphql server (Navigate to the server's root directory in Terminal, and run `npm start`)
7. Tap the 'Generate Character` button twice
8. Note that the app labels are now populated
9. Tapping the 'Generate Character' button repeatedly should populate different class names in the `Class:` label


## Any background info you would like to include :white_check_mark:
Currently, only the `Class` data is returned from the D&D API. The `Race:` and `Alignment` labels are temporarily populated with `Half-Orc` and `Chaotic Neutral` respectively, until the schema and query is updated to include other results.


## Screenshots (if applicable) :camera:
![dndgenerator-pr-create-ui](https://user-images.githubusercontent.com/8409475/57650540-0a9cb900-7599-11e9-88a4-35e8c2f1f6ba.gif)
